hack/local-up-cluster.sh

```diff
@@ -688,111 +687,113 @@ function start_kubelet {
 
     priv_arg=""
     if [[ -n "${ALLOW_PRIVILEGED}" ]]; then
-      priv_arg="--allow-privileged=${ALLOW_PRIVILEGED} "
+      priv_arg="--allow-privileged=${ALLOW_PRIVILEGED}"
     fi
 
-    cloud_config_arg="--cloud-provider=${CLOUD_PROVIDER} --cloud-config=${CLOUD_CONFIG}"
+    cloud_config_arg=("--cloud-provider=${CLOUD_PROVIDER}" "--cloud-config=${CLOUD_CONFIG}")
     if [[ "${EXTERNAL_CLOUD_PROVIDER:-}" == "true" ]]; then
-       cloud_config_arg="--cloud-provider=external"
-       cloud_config_arg+=" --provider-id=$(hostname)"
+       cloud_config_arg=("--cloud-provider=external")
+       cloud_config_arg+=("--provider-id=$(hostname)")
     fi
 
     mkdir -p "/var/lib/kubelet" &>/dev/null || sudo mkdir -p "/var/lib/kubelet"
     # Enable dns
     if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
       if [[ "${ENABLE_NODELOCAL_DNS:-}" == "true" ]]; then
-        dns_args="--cluster-dns=${LOCAL_DNS_IP} --cluster-domain=${DNS_DOMAIN}"
+        dns_args=("--cluster-dns=${LOCAL_DNS_IP}" "--cluster-domain=${DNS_DOMAIN}")
```

Comment:
Confirming my understanding is correct - the `()` is added because we want to quote `--cluster-dns...` and `--cluster-domain=...` separately?

Does shellcheck complain if we quote them together?

Reply:
This is a change to split into each option.
dns_args have to be quoted for fixing the following shellcheck message.

In ./hack/local-up-cluster.sh line 774:
      ${dns_args}
      ^---------^ SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.

If dns_args not defined as an array, it was executed as one option.

e.g.
hyperkube kubelet --allow-privileged=true ... '--cluster-dns=10.0.0.10 --cluster-domain=cluster.local' port=10250

Reply:
:+1: